trunk/source/SMTSolverBridge/src/de/uni_freiburg/informatik/ultimate/smtsolver/external/scriptResult.cup

```diff
@@ -580,3 +584,23 @@ goResponse ::= TRUE {: RESULT = true; :}
 			 | STDOUT:s {: RESULT = s; :}
 			 | STDERR:s {: RESULT = s; :}
 			 | errorResponse;
+
+/* get-model response, currently only for Horn solvers */
+gmResponse ::= LPAR funDef*:d {: RESULT = new ModelDescription(Set.of(d)); :} RPAR;
+
+funDef ::= LPAR DEFINEFUN {: localVars.beginScope(); :} symbol:fun
+                      LPAR sortedVar*:vars RPAR sort:resultSort term:body RPAR
+            {: localVars.endScope();
+               try {
+                       // TODO Support definition of auxiliary functions
```

Comment:
Just a minor comment: Support for auxiliary functions should be also possible (requires some work of course) in a similar way, right?

Reply:
Or perhaps one could directly inline the auxiliaries?

Let's see if this poses a problem in practice first.